Use ember-modifier v3 API

tcjr opened the pull request:
This PR updates the class-based modifiers to use the ember-modifier v3 API.

In summary
- The lifecycle hooks `didInstall`, `didReceiveArguments`, `didUpdateArguments` have all been replaced with a single `modify` hook. This hook gets called on initial install _and_ whenever the args change.
- The `Modify` base class no longer provides `this.element`. Instead, `element` is provided in the `modify` hook, so the modifier implementation sets `this.element` there.
- The `Modify` base class no longer provides `this.args`. Instead, positional and named args are passed to the `modify` hook.  I updated the implementation to set `this._namedArgs` and `this._positionalArgs` so the rest of the code can use those where it used to use `this.args.named` and `this.args.positional` respectively.
- The `willDestroy` hook has been removed.  Instead, the implementation registers a cleanup function in the modifier's constructor using `@ember/destroyable`.  I moved the code that was in the `willDestroy` hook into this cleanup function.

I updated the ember-try scenarios and ci to include the last two LTS versions (4.4 and 4.8).

I also updated the `package.json` to drop `^2.0.0` from the `ember-modifier` dependency.  I guess this means if you merge this, it's a breaking change so maybe you'll need to bump the major version.

Fixes #4

tcjr committed:
Updated modifiers to use ember-modifier v3 API

donaldsonjulia commented:
Would love if this could be merged. At this point we cannot continue to use `ember-tippy` in our project because of the `ember-modifier` deprecations. The changes made in this PR fix the functionality.

nag5000 commented:
@tcjr The changes of package-lock.json and .npmrc seems are not really related to this PR, could you exclude it from the changes?

I no longer follow the ember-modifier changes and the ember ecosystem in general, but I believe I can merge it and release it as a beta version. Thank you!

tcjr committed:
Removed .npmrc and updated npm build options in ember-try config.

tcjr committed:
Updated ci to use node 16

tcjr committed:
Updated package-lock for ci to build

tcjr committed:
Merge pull request #1 from LAWPRCT/remove-npmrc

Removed .npmrc and updated npm build options in ember-try config.
Bumped github actions to use Node 16. See: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

tcjr commented:
@nag5000 I removed the .npmrc and instead updated the ember-try config.  I also updated the CI workflow since [github wants us to use Node 16 in actions](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/).
